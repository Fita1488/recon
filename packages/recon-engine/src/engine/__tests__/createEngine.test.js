const expect = require('expect');
const Path = require('path');
const FS = require('fs');

const createEngine = require('../createEngine');

function run(path) {
  const absPath = Path.join(__dirname, '../__fixtures__', path);
  const output = require(Path.join(absPath, 'output'));
  const query = FS.readFileSync(Path.join(absPath, 'query.graphql'), {encoding: 'utf8'});

  const engine = createEngine({
    files: '**/*.js',
    cwd: Path.join(absPath, 'src')
  });

  return new Promise((accept, reject) => {
    setTimeout(() => { // TODO: Need some way of knowing when recon is working/ready :D
      engine.runQuery(query).then(
        result => {
          expect(result).toMatch(output);
          accept();
        }
      );
    }, 1000);
  })
}

describe('react-engine::engine/createEngine', () => {
  describe('::createEngine (default)', () => {

    it('should respond to query as expected: basic-app', () => run('basic-app'));

  });
});
