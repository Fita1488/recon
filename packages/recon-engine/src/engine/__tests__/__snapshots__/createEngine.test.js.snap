// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should respond to query as expected: basic-app 1`] = `
Object {
  "errors": Array [
    [GraphQLError: Syntax Error GraphQL request (1:2) Expected Name, found }

1: {}
    ^
2: 
],
  ],
}
`;
